authors/apps/authentication/tests/test_user_registration.py

```diff
@@ -0,0 +1,119 @@
+"""
+Module to test functionality to register a user
+"""
+from rest_framework import status
+from rest_framework.test import APITestCase, APIClient
+
+
+class TestUserRegistration(APITestCase):
+    """
+    class to handle user registraotion tests
+    """
+
+    def setUp(self):
+        """
+         Method to handle data to be setup before running tests
+        """
+        self.client = APIClient()
+        self.no_data = {}
+        self.data =  {
+            "user": {
+                "username": "user",
+                "email": "testuser@gmail.com",
+                "password": "user@12345"
+            }
+        }
+        self.data1 = {
+                "user": {
+                    "username": "",
+                    "email": "userstest67@gmail.com",
+                    "password": "Users@12345"
+            }
+        }
+        self.data2 = {
```

Comment:
@felixkiryowa kindly use more descriptive names for your variables. data1....n would be hard when onboarding to a larger codebase. Try something like "user_post_data_1_test".

Reply:
Let me rectify it